chain/index/ddls_test.go

```diff
@@ -0,0 +1,708 @@
+package index
+
+import (
+	"database/sql"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestHasRevertedEventsInTipsetStmt(t *testing.T) {
+	s, err := NewSqliteIndexer(":memory:", nil, 0, false, 0)
+	require.NoError(t, err)
+
+	// running on empty DB should return false
+	verifyHasRevertedEventsInTipsetStmt(t, s, []byte("test_tipset_key"), false)
+
+	// Insert tipset with a reverted event
+	ts := tipsetMessage{
+		tipsetKeyCid: []byte("test_tipset_key"),
+		height:       1,
+		reverted:     false,
+		messageCid:   []byte("test_message_cid"),
+		messageIndex: 0,
+	}
+	messageID := insertTipsetMessage(t, s, ts)
```

Comment:
Suggestion (I'm aware of the amount of additional work involved so it's up to you): add some slop in the DB that you expect _should not_ be modified by your updates and confirm that your update is targeted. As it is, this test is going to pass even if your `UPDATE` didn't have a `WHERE` clause.

Reply:
This is done. We now also insert a reverted event and make sure it's un-reverted till the end.